server/db/driver/pg/matches.go

```diff
@@ -75,34 +78,60 @@ func userMatches(ctx context.Context, dbe *sql.DB, tableName string, aid account
 	return ms, nil
 }
 
-// ActiveMatches retrieves a UserMatch slice for active matches involving the
-// given user. Swaps that have successfully completed or failed are not
-// included.
-func (a *Archiver) ActiveMatches(aid account.AccountID) ([]*order.UserMatch, error) {
+// AllActiveUserMatches retrieves a MatchData slice for active matches in all
+// markets involving the given user. Swaps that have successfully completed or
+// failed are not included.
+func (a *Archiver) AllActiveUserMatches(aid account.AccountID) ([]*db.MatchData, error) {
 	ctx, cancel := context.WithTimeout(a.ctx, a.queryTimeout)
 	defer cancel()
 
-	var matches []*order.UserMatch
+	var matches []*db.MatchData
 	for m := range a.markets {
 		matchesTableName := fullMatchesTableName(a.dbName, m)
-		matchesM, err := activeUserMatches(ctx, a.db, matchesTableName, aid)
+		mdM, err := userMatches(ctx, a.db, matchesTableName, aid, false)
 		if err != nil {
 			return nil, err
 		}
-		matches = append(matches, matchesM...)
+
+		matches = append(matches, mdM...)
 	}
 
 	return matches, nil
 }
 
-func activeUserMatches(ctx context.Context, dbe *sql.DB, tableName string, aid account.AccountID) ([]*order.UserMatch, error) {
+// ActiveMatches retrieves a UserMatch slice for active matches involving the
+// given user. Swaps that have successfully completed or failed are not
+// included. Assigned swap contract txn fee rates for both base and quote assets
+// for each swap are also returned, one of which corresponds to the FeeRateSwap
+// of the corresponding UserMatch.
+// TODO: Remove.
+func (a *Archiver) ActiveMatches(aid account.AccountID) (matches []*order.UserMatch, matchTimes []time.Time, baseFeeRates, quoteFeeRates []uint64, err error) {
+	ctx, cancel := context.WithTimeout(a.ctx, a.queryTimeout)
+	defer cancel()
+
+	for m := range a.markets {
+		matchesTableName := fullMatchesTableName(a.dbName, m)
+		matchesM, matchTimesM, baseRatesM, quoteRatesM, err0 := activeUserMatches(ctx, a.db, matchesTableName, aid)
+		if err0 != nil {
+			return nil, nil, nil, nil, err0
+		}
+		matches = append(matches, matchesM...)
+		matchTimes = append(matchTimes, matchTimesM...)
+		baseFeeRates = append(baseFeeRates, baseRatesM...)
+		quoteFeeRates = append(quoteFeeRates, quoteRatesM...)
+	}
+
+	return
+}
+
+func activeUserMatches(ctx context.Context, dbe *sql.DB, tableName string, aid account.AccountID) (ms []*order.UserMatch, matchTimes []time.Time, baseFeeRates, quoteFeeRates []uint64, err error) {
```

Comment:
Both of these functions can probably be removed.  ActiveMatches is no longer an interface function or used anywhere.